### DISCUSSION TITLE
can't install racket packages during formula install?

### DISCUSSION THREAD

**Discussion by aryairani** [Homebrew/discussions#5275] [Writing Formulae/Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.17-37-g7c759d3
ORIGIN: https://github.com/Homebrew/brew
HEAD: 7c759d3eade70ad66f7083f9176759dd0af04cce
Last commit: 9 minutes ago
Core tap HEAD: 7671113ad6c67e1fa730f05cb40b5612ab475e0b
Core tap last commit: 9 weeks ago
Core tap branch: topic/unison-local-ui
Core tap JSON: 10 Apr 05:37 UTC
Core cask tap JSON: 10 Apr 05:37 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.GC2z5o3shL/org.xquartz:0
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.43.0 => /opt/homebrew/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.4.1-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: 15.2
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  exa
  minimal-racket
  openssl@1.1
  vault

Warning: Unbrewed dylibs were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected dylibs:
  /usr/local/lib/libcpsrt.dylib

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  unisonlang

Warning: You have uncommitted modifications to Homebrew/homebrew-core.
If this is a surprise to you, then you should stash these modifications.
Stashing returns Homebrew to a pristine state but can be undone
should you later need to do so for some reason.
  cd /opt/homebrew/Library/Taps/homebrew/homebrew-core && git stash -u && git clean -d -f

Uncommitted files:
  ?? .DS_Store
  ?? Formula/.DS_Store
  ?? Formula/u/.unisonHistory

Warning: Some taps are not on the default git origin branch and may not receive
updates. If this is a surprise to you, check out the default branch with:
  git -C $(brew --repo homebrew/core) checkout master

Warning: /usr/bin occurs before /opt/homebrew/bin in your PATH.
This means that system-provided programs will be used instead of those
provided by Homebrew. Consider setting your PATH so that
/opt/homebrew/bin occurs before /usr/bin. Here is a one-liner:
  echo 'export PATH="/opt/homebrew/bin:$PATH"' >> ~/.zshrc

The following tools exist at both paths:
  bashbug
  git
  git-receive-pack
  git-shell
  git-upload-archive
  git-upload-pack
  openssl
  pip3
  python3
```

### Description of issue

I'm trying to write a formula that builds a Racket program. It requires a racket library installed, which is normally done with `raco pkg install <path to library>`

If I run this command from within `def install`, I get an "Operation not permitted" error.
```
open-output-file: error opening file
  path: /opt/homebrew/opt/minimal-racket/share/racket/pkgs/.LOCKpkgs.rktd
  system error: Operation not permitted; errno=1
  context...:
```
If I run it from the shell outside of `brew`, it succeeds.

1) I'm not sure why it would be different inside `brew` vs outside, any ideas?

2), the available racket formula `minimal-racket` is deprecated due to relying on a deprecated openssl.
There is a Cask for racket, but I don't think I can install a cask from a formula, is that right?

I guess that's it for now; thanks in advance.

**Comment by SMillerDev** [comment]
> I'm not sure why it would be different inside brew vs outside, any ideas?

Brews use of the macOS sandbox to prevent software writing in other software's installation.
> the available racket formula minimal-racket is deprecated due to relying on a deprecated openssl.
> There is a Cask for racket, but I don't think I can install a cask from a formula, is that right?

Not in Homebrew/core. But in a tap you can do whatever you want.

**Reply by aryairani** [reply]
Hi @SMillerDev, thanks—

> Brews use of the macOS sandbox to prevent software writing in other software's installation.

I see. Is there a Brew-approved way for installing a racket library needed to build a formula? Or can I help invent one?

> > There is a Cask for racket, but I don't think I can install a cask from a formula, is that right?

> Not in Homebrew/core. But in a tap you can do whatever you want.

I couldn't find docs or syntax on this, could you point me in the right direction? But also, I imagine I would run into the same sandbox restriction?

**Reply by aryairani** [reply]
Sorry, I mean I found `depends_on :cask => "osxfuse"` syntax, but I get this error:
```
Unsupported special dependency :cask
```

**Reply by SMillerDev** [reply]
> Is there a Brew-approved way for installing a racket library needed to build a formula?

No clue about racket, but if you can install this library outside the Racket installation with some environment variables or something. That would work.

**Reply by SMillerDev** [reply]
> I couldn't find docs or syntax on this, could you point me in the right direction? 

I believe `depends_on "homebrew/cask/racket"` should just work. I put the full name there to be sure, it might also work with only racket.

> But also, I imagine I would run into the same sandbox restriction?

Homebrew generally has no idea where casks install so it doesn't protect them

**Reply by aryairani** [reply]
>I believe depends_on "homebrew/cask/racket" should just work. I put the full name there to be sure, it might also work with only racket.

Ah hm, based on https://github.com/orgs/Homebrew/discussions/3163 (where you had previously weighed in), I didn't try it.

But with your latest message, I did try it; but it didn't work for me:

`depends_on "homebrew/cask/racket"` gave 
```
Error: No available formula with the name "homebrew/cask/racket".
Please tap it and then try again: brew tap homebrew/cask
```

And `brew tap homebrew/cask`, which I'd rather avoid the users having to do, gave
```
Error: Tapping homebrew/cask is no longer typically necessary.
Add --force if you are sure you need it done.
```

And `brew tap homebrew/cask --force` didn't seem to help with the original `depends_on`.

Thanks for the help, but do you have any other ideas I could try? Or pointers to docs.

P.S. `depends_on racket` gives
```
Warning: Formula racket was renamed to minimal-racket.
==> Downloading https://formulae.brew.sh/api/cask.jws.json

Warning: Treating racket as a formula. For the cask, use homebrew/cask/racket or specify the `--cask` flag.
Warning: minimal-racket has been deprecated because it uses deprecated `openssl@1.1`!
==>
```
